rules/ssrf.go

```diff
@@ -0,0 +1,70 @@
+package rules
+
+import (
+	"go/ast"
+	"go/types"
+
+	"github.com/securego/gosec"
+)
+
+type ssrf struct {
+	gosec.MetaData
+	gosec.CallList
+}
+
+// ID returns the identifier for this rule
+func (r *ssrf) ID() string {
+	return r.MetaData.ID
+}
+
+// ResolveVar tries to resolve the first argument of a call expression
+// The first argument is the url
+func (r *ssrf) ResolveVar(n *ast.CallExpr, c *gosec.Context) bool {
+	if len(n.Args) > 0 {
+		arg := n.Args[0]
+		if ident, ok := arg.(*ast.Ident); ok {
+			obj := c.Info.ObjectOf(ident)
+			if _, ok := obj.(*types.Var); ok && !gosec.TryResolve(ident, c) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// Match inspects AST nodes to determine if certain net/http methods are called with variable input
+func (r *ssrf) Match(n ast.Node, c *gosec.Context) (*gosec.Issue, error) {
+	// Call expression is using http package directly
+	if node := r.ContainsCallExpr(n, c); node != nil {
+		if r.ResolveVar(node, c) {
+			return gosec.NewIssue(c, n, r.ID(), r.What, r.Severity, r.Confidence), nil
+		}
+	}
+	// Look at the last selector identity for methods matching net/http's
+	if node, ok := n.(*ast.CallExpr); ok {
```

Comment:
If I understand correctly, `GetCallInfo` returns the package or type of a call expression. The first check will use it to find all `net/http` functions (eg. `http.Get(url), http.Post(url)`).  This second check is meant to flag function calls that are not from `net/http` but have the same ending `net/http` method signatures (eg. `resty.R().Get(url)` or `someCustomStruct.Post(url)`). To do this, it can only look at the ending selector's identity to see if its matches a Get/Post/Put etc and therefore cannot use `GetCallInfo`.